instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/SpringWebMvcTracer.java

```diff
@@ -110,6 +119,18 @@ private void onRender(SpanBuilder span, ModelAndView mv) {
     }
   }
 
+  @Override
+  protected Throwable unwrapThrowable(Throwable throwable) {
+    if (throwable instanceof InvocationTargetException) {
+      return throwable.getCause();
+    }
+    return super.unwrapThrowable(throwable);
+  }
+
+  public void addUnwrappedThrowable(Span span, Throwable throwable) {
+    addThrowable(span, unwrapThrowable(throwable));
+  }
```

Comment:
I think it would be reasonable for addThrowable to always call unwrapThrowable, similar to endExceptionally (current impl is good for this PR) cc: @mateuszrzeszutek

Reply:
nm, just reviewing #2513 now and see this is already happening there 👍